middlewares.go

```diff
@@ -45,7 +45,7 @@ func scrubInsertionIDs(ctx *ExecutionContext, response map[string]interface{}) e
 	for field, locations := range ctx.Plan.FieldsToScrub {
 		for _, location := range locations {
 			// look for the insertion points in the response for the field
-			insertionPoints, err := executorFindInsertionPoints(&lock, location, ctx.Plan.Operation.SelectionSet, response, [][]string{})
+			insertionPoints, err := executorFindInsertionPoints(&lock, location, ctx.Plan.Operation.SelectionSet, response, [][]string{[]string{}}, ctx.Plan.FragmentDefinitions)
```

Comment:
This invocation of `executorFindInsertionPoints` was returning a `[][]string{}` value (i.e. no insertion points). This resulted in the "id" field not to be scrubbed from the response.

The only other invocation of this function received a `[][]string{insertionPoints}` with `insertionPoints = []string{}`. By inspecting the code in `executorFindInsertionPoints` and seeing that, for a `[][]string{}` value, the loop to append points to `oldBranch` never runs, I decided to make this match by passing in a `[][]string{[]string{}}` value.

This has fixed the scrubbing of id problems. Unsure why this was not an issue for non-fragment federation calls.